### PR TITLE
Cleanup @property stuff on BaseRemoteEndpoint

### PR DESCRIPTION
**Pull request by pipermerriam**
## What was wrong?

There were some errors in the implementation that needed cleaning up

## How was it fixed?

Fixed them (need to add tests)

#### Cute Animal Picture

6a010535647bf3970b0105367c839d970b-800wi

### CODE REVIEW

File: lahja/base.py [hunk 1]
```diff
@@ -221,12 +221,15 @@ def name(self) -> str:
     async def wait_stopped(self) -> None:
         await self._stopped.wait()
 
+    @property
     def is_running(self) -> bool:
-        return not self.is_stopped and self.running.is_set()
+        return not self.is_stopped and self._running.is_set()
```
Comment (cburgdorf) [inline lahja/base.py]: I'm really confused by this. How is it possible that neither `mypy` nor `flake8` caught this?

Comment (pipermerriam) [reply lahja/base.py]: In my local env. just pushed something to see if CI catches it.

Comment (pipermerriam) [reply lahja/base.py]: it did https://circleci.com/gh/pipermerriam/lahja/2245

Comment (cburgdorf) [reply lahja/base.py]: Yes, it seems it catches it *after* this change but not on current master which is really weird. Also, the `@property` decorators also need to be applied on the `RemoteEndpointAPI`.

Comment (cburgdorf) [reply lahja/base.py]: In fact, as soon as you remove the `@property` from `is_stopped` it won't catch this error anymore. This feels like a weird mypy bug to me.

Comment (pipermerriam) [reply lahja/base.py]: That's very odd.

Comment (cburgdorf) [reply lahja/base.py]: Ok, here is a minimal test case.

```python
class TestCase:

    def foo(self) -> bool:
        return False and self.meh()
```

This is not caught by `mypy`. Do you think this is expected behavior?

Comment (cburgdorf) [reply lahja/base.py]: I opened an issue against mypy https://github.com/python/mypy/issues/6990

**Pull request merged**